src/test/java/org/jenkinsci/plugins/pipeline/utility/steps/jenkins/NodesByLabelStepTest.java

```diff
@@ -53,17 +53,23 @@
     @Before
     public void setup() throws Exception {
         job = r.jenkins.createProject(WorkflowJob.class, "workflow");
-        r.createSlave("dummy1", "a", new EnvVars());
-        r.createSlave("dummy2", "a b", new EnvVars());
-        r.createSlave("dummy3", "a b c", new EnvVars());
-        DumbSlave slave = r.createSlave("dummy4", "a b c d", new EnvVars());
-        slave.toComputer().waitUntilOnline();
+
+        DumbSlave[] dummies = {
+            r.createSlave("dummy1", "a", new EnvVars()),
+            r.createSlave("dummy2", "a b", new EnvVars()),
+            r.createSlave("dummy3", "a b c", new EnvVars()),
+            r.createSlave("dummy4", "a b c d", new EnvVars()),
+        };
+        for (DumbSlave dummy : dummies) {
+            dummy.toComputer().waitUntilOnline();
```

Comment:
These tests were pretty flaky on the Windows agents. I believe it was due to a race condition since it was only waiting for the last dummy to come online.